kubernetes/linux/Dockerfile

```diff
@@ -18,6 +16,10 @@ ENV KUBE_CLIENT_BACKOFF_DURATION 0
 ENV RUBY_GC_HEAP_OLDOBJECT_LIMIT_FACTOR 0.9
 RUN /usr/bin/apt-get update && /usr/bin/apt-get install -y libc-bin wget openssl curl sudo python-ctypes init-system-helpers  net-tools rsyslog cron vim dmidecode apt-transport-https gnupg && rm -rf /var/lib/apt/lists/*
 COPY setup.sh main.sh defaultpromenvvariables defaultpromenvvariables-rs defaultpromenvvariables-sidecar mdsd.xml envmdsd logrotate.conf $tmpdir/
+

```

Comment:
Moving these 2 to higher layers makes it faster ?

Reply:
yeah, assuming you use a new image tag every time you build an image (which at least I always do so I can be sure I'm testing my latest build).  The problem is that whenever AGENT_TAG changes it invalidates all cached layers before it. By moving the IMAGE_TAG arg as far down as possible it will invalidate as few cached layers as possible

Reply:
that makes sense, but then to capitalize on that you need to be building multiple builds/images in the same node (like you dev node). It wont make much difference on build containers as these layers are missing there every single time we build .

Reply:
I completely agree, it only helps for repeated builds (like on my dev machine). It really helps when repeatedly building images though. 

For example, yesterday I built ~10 dev images. Each one took about 5 minutes, so that's a lot of wasted time. This change would have brought each local build down to threeish minutes.

Reply:
ok. thanks David. Also there are ways to remove this image creation time tag passing to be used for our telemetry, which will completely remove these 2 (ARG & ENV). I will resolve this comment.